src/utils/hooks/useGetResponsesByCode.tsx

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from 'react'
+import {
+  Application,
+  ApplicationResponse,
+  ApplicationSection,
+  useGetApplicationQuery,
+} from '../generated/graphql'
+import { ResponsesByCode } from '../types'
+
+interface useLoadApplicationProps {
+  serialNumber: string
+}
+
+const useGetResponsesByCode = (props: useLoadApplicationProps) => {
+  const { serialNumber } = props
+  const [responsesByCode, setResponsesByCode] = useState({})
+  const { data, loading, error } = useGetApplicationQuery({
+    variables: { serial: serialNumber },
+  })
+
+  useEffect(() => {
+    if (data?.applications) {
+      if (data.applications.nodes.length === 0) return
+      if (data.applications.nodes.length > 1)
+        console.log('More than one application returned. Only one expected!')
+
+      const applicationResponses = data.applications.nodes[0]?.applicationResponses
+        .nodes as ApplicationResponse[]
+
+      const currentResponses = {} as ResponsesByCode
+
+      applicationResponses.forEach((response) => {
+        const code = response?.templateElement?.code
+        if (code) currentResponses[code] = response?.value?.text || response?.value
+      })
+
+      setResponsesByCode(currentResponses)
+    }
+  }, [data, error])
+
+  return {
+    error,
+    loading,
+    responsesByCode,
+  }
+}
+
+export default useGetResponsesByCode
```

Comment:
I would recommend using a similar code to what was suggested by Andrei on branch [27-suggested-improvements](https://github.com/openmsupply/application-manager-web-app/blob/%2327-suggested-improvements/src/utils/hooks/useLoadTemplate.tsx). 
Basically renaming the variables `error` to `apolloError` and `loading`to `apolloLoading` (see my previous suggestions). Then keeping track of each unexpected problems found (more than one application found) to the **new** `error` variable. Which can be done in a separated function (check Andrei suggestion). Also the loading should have another one that is handled by us and is set to `true` only after the object is filled or if an error occurred.

Reply:
Okay, think I've covered it. Please let me know if it's correct.